js/src/UXCapture.js

```diff
@@ -33,11 +33,15 @@ export default class UXCapture {
     });
   }
 
-  mark(name) {
+  mark(name, waitForNextPaint = true) {
     const mark = ExpectedMark.get(name);
 
     if (mark) {
-      mark.record();
+      if (waitForNextPaint) {
+        mark.waitForNextPaintAndRecord();
+      } else {
+        mark.record();
```

Comment:
probably worth adding a code comment block for this method to explain in which cases a caller would _not_ want to `waitForNextPaint`